BREAKING CHANGE(hubble): introduce new version V2

FrostyHec opened the pull request:
## 📌 Background & Motivation

Apache HugeGraph Hubble 2.0 is a major platform evolution aimed at modernizing the web UI, streamlining backend services, aligning core client drivers, and establishing robust continuous integration gates. 

This PR consolidates **370+ fragmented historic commits into a linear sequence of 47 clean, Conventional-compliant commits**. It purges legacy modules, introduces React + TS foundations, stabilizes E2E automation pipelines, and enforces license and quality metrics.

---

## 🛠️ Key Architectural Changes

The integration flow of Hubble 2.0 components can be represented below:

```mermaid
graph TD
    subgraph "Hubble 2.0 Web UI (hubble-fe)"
        FE_Pages["React + TS UI Components <br>(i18n / canvas / scheduler)"]
    end

    subgraph "Hubble 2.0 Backend (hubble-be)"
        BE_Controller["MVC Controllers <br>(auth / algorithm / task)"]
        BE_Service["GraphsService / SchemaService"]
        BE_H2["Metadata Database (H2)"]
    end

    subgraph "Client Layer (hugegraph-client)"
        HG_Client["HugeClient (BuilderFacade)"]
    end

    subgraph "Core Graph Engine"
        HG_Server["HugeGraph Server"]
    end

    %% Flow Relationships
    FE_Pages -- "Rest API Requests" --> BE_Controller
    BE_Controller -- "Persist State" --> BE_H2
    BE_Controller -- "Manage & Coordinate" --> BE_Service
    BE_Service -- "Interact" --> HG_Client
    HG_Client -- "Gremlin / Schema API" --> HG_Server
```

---

## 📝 Major Change Dimensions

### 1. Frontend Retirement & Modernization (`hubble-fe`)
* **Legacy Cleanups**: Purged the obsolete `hubble-fe-old` directories (~50k lines of legacy code retired) to reduce package sizes and maintenance overhead.
* **i18n & Localization**: Standardized internationalization properties (`zh-CN` / `en-US`) across core workspaces (Olap components, import scheduler, vertex/edge modals, task panels).
* **UI Stabilization**: Refactored topological canvas controls, fixed double-triggering modal state resets during datasource creation, and localized canvas tooltips.

### 2. Backend Restructuring & API Alignment (`hubble-be`)
* **Facade Simplification**: Re-aligned `GraphsService` interface signatures to rely strictly on the standard `HugeClient` API instead of outdated custom wrappers.
* **Security & Auth**: Hardened role-authorization configurations to guard protected endpoints, supporting SaaS integrations, LangChain orchestration parameters, and Operator environments.
* **Data Safety**: Implemented strict, explicit confirmation gates for all destructive graph operations (such as graph clears and async drop tasks).

### 3. CI/CD Pipeline & Quality Gates
* **Checkstyle Enforced**: Re-enabled strict style compliance gates (`tools/checkstyle.xml`) to prevent format regressions on backend source files.
* **License Audit**: Restored complete license header compliance audits for all build targets.
* **Automated E2E Testing**: Established fully automated Playwright browser smoke tests running against local test containers.

---

## 🚀 Execution Logic: Local E2E Smoke Testing Workflow

The diagram below outlines the automated validation pipeline executed before release packaging:

```
[GitHub CI / Local JDK11]
           │
           ▼ (mvn package -P unit-test)
┌────────────────────────────────────────┐
│      1. Compile & Checkstyle Audit      │
└────────────────────────────────────────┘
           │
           ▼ (Install Playwright Chromium)
┌────────────────────────────────────────┐
│      2. Start Hubble Server & H2       │ ◄── [Set active session to 48h]
└────────────────────────────────────────┘
           │
           ▼
┌────────────────────────────────────────┐
│   3. Spin up HugeGraph Server (Auth)   │
└────────────────────────────────────────┘
           │
           ▼ (Inject Mock Graphs)
┌────────────────────────────────────────┐
│   4. Run Playwright E2E Test Suite     │
│      - Create connection               │
│      - Execute Gremlin shortest-path   │
│      - Verify UI graph topology canvas │
└────────────────────────────────────────┘
           │
           ▼ (Teardown)
[Clean System Containers]
```

---

## 🔍 Verification & Evidence Reference

* **Compiles Successfully**: Checked local build consistency:
  ```bash
  mvn clean install -DskipTests -Dmaven.javadoc.skip=true -ntp
  ```
* **Contract Tests**: Verified `SessionTimeoutConfigTest` for session token expiration and `SchemaServiceViewTest` for schema controller assertions:
  ```bash
  mvn test -P unit-test -pl hugegraph-hubble/hubble-be -ntp
  ```
* **Git Tree Cleanliness**: Verified no codebase divergence against current `hubble2` branch:
  ```bash
  git diff hubble2 github-desktop-FrostyHec/hubble2.0
  ```
  *(Returned empty diff, confirming complete parity)*

FrostyHec committed:
feat(hubble-fe): migrate frontend assets

FrostyHec committed:
frontend transfer

FrostyHec committed:
update i18n configuration

FrostyHec committed:
i18n configuration on components and algorithm form, algorithm search

FrostyHec committed:
supporting i18n on olap components

FrostyHec committed:
fix(hubble-fe): support i18n on olap components

FrostyHec committed:
Update exist class in backend to support new version.

FrostyHec committed:
support i18n for mode switch text

FrostyHec committed:
support auth api in hubble-backend

FrostyHec committed:
feat(hubble): support auth api in hubble-backend

FrostyHec committed:
support algorithm api in hubble-backend

FrostyHec committed:
feat(hubble): support algorithm api in hubble-backend

FrostyHec committed:
feat(hubble): support saas, langchain and op in hubble-backend

FrostyHec committed:
supporting saas, langchain and op in hubble-backend

FrostyHec committed:
refactor(hubble): update files to support hubble-2.0 backend

FrostyHec committed:
updating files to support hubble-2.0 backend

FrostyHec committed:
fix complication error from i18n

FrostyHec committed:
fix dependency

FrostyHec committed:
fix i18n display

FrostyHec committed:
update license header

FrostyHec committed:
update license header

FrostyHec committed:
update file format

FrostyHec committed:
update file format

FrostyHec committed:
update path

FrostyHec committed:
remove common-logger

FrostyHec committed:
recover legacy version

FrostyHec committed:
Merge branch 'refs/heads/master' into hubble2.0

FrostyHec committed:
use hg-common-1.3.0

Thespica committed:
add GraphSpace related structres and APIs

Thespica committed:
add GraphSpace related structres and APIs

Thespica committed:
add enums for HugeType

Thespica committed:
fix gs for client, remove unused manager

Thespica committed:
fix gs for client, remove unused manager

Thespica committed:
adapt more gs-related field and constructor, and recoded them

Thespica committed:
adapt more gs-related field and constructor, and recoded them

Thespica committed:
adapt gs for api/schema/*

Thespica committed:
adapt gs for some apis and tests

Thespica committed:
adapt gs for some apis and tests

Thespica committed:
modify ci to enable gs-filter

Thespica committed:
modify ci to enable gs-filter

Thespica committed:
update copyright

Thespica committed:
update copyright

Thespica committed:
enable server gs-filter

Thespica committed:
enable server gs-filter

Thespica committed:
ci run all tests

Thespica committed:
ci run all tests

Thespica committed:
refine ci test

Thespica committed:
refine ci test

Thespica committed:
refine ci server script

Thespica committed:
refine ci server script

Thespica committed:
refine ci rest conf

Thespica committed:
refine ci rest conf

Thespica committed:
basic support gs filter

Thespica committed:
remove redundante null parameter

Thespica committed:
RestClient record core version

Thespica committed:
get supportGs by core version

Thespica committed:
close gs filter option

Thespica committed:
close gs filter option

Thespica committed:
fix test to let client created by clone

Thespica committed:
fix test to let client created by clone

Thespica committed:
fix params order when building client

Thespica committed:
stop ci if test failed

Thespica committed:
stop ci if test failed

Thespica committed:
add default overload of graph space for client builder

Thespica committed:
try update server's commit of loader-ci and sportk-connector-ci

Thespica committed:
try update server's commit of loader-ci and sportk-connector-ci

Thespica committed:
align default graph space name to DEFAULT(UPPER CASE)

Thespica committed:
restore format of rest-server.properties

Thespica committed:
restore format of rest-server.properties

Thespica committed:
restore ci server commit id

Thespica committed:
align parameter name of test

Thespica committed:
remove zh-ch documentation

Thespica committed:
add todo about be compatitable to server do not support gs

imbajin committed:
Merge branch 'master' into client-support-gs

FrostyHec committed:
update yarn

FrostyHec committed:
fix dependency

FrostyHec committed:
complete merging

FrostyHec committed:
fixing dependency

FrostyHec committed:
Merge branch 'refs/heads/pr-633' into hubble2.0

Thespica committed:
add missing classes in api

Thespica committed:
add missing classes in api

Thespica committed:
add missing classes in structure

Thespica committed:
add missing classes in structure

Thespica committed:
add missing classes in driver

Thespica committed:
add missing classes in driver

Thespica committed:
add gs parameter for traversers and related APIs

Thespica committed:
add gs parameter for traversers and related APIs

Thespica committed:
adjust neighborRank and personalRank to return RanksWithMeasure

Thespica committed:
adjust neighborRank and personalRank to return RanksWithMeasure

Thespica committed:
adjust HugeClient for whiteIpListManager

Thespica committed:
adjust HugeClient for whiteIpListManager

Thespica committed:
add api/ip/WhiteIpListAPI and adjust related import

Thespica committed:
add api/ip/WhiteIpListAPI and adjust related import

Thespica committed:
add pd-client1.5.0, pd-grpc1.5.0

Thespica committed:
add pd-client1.5.0, pd-grpc1.5.0

Thespica committed:
add WhiteIpListAPI and adjust treversser

Thespica committed:
add WhiteIpListAPI and adjust treversser

Thespica committed:
upgrade java to 11 for client ci

Thespica committed:
upgrade java to 11 for client ci

Thespica committed:
upgrade mockito version for test with java 11

Thespica committed:
upgrade mockito version for test with java 11

Thespica committed:
modify deserializtion of RanksWithMeasure

Thespica committed:
modify deserializtion of RanksWithMeasure

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need' into hubble2.0

Thespica committed:
add structure/traverser/Egonet and structure/traverser/PathWithMeasure

Thespica committed:
add structure/traverser/Egonet and structure/traverser/PathWithMeasure

Thespica committed:
modify structure/auth

Thespica committed:
modify structure/auth

Thespica committed:
edit auth related API

Thespica committed:
edit auth related API

Thespica committed:
adjust driver/AuthManager

Thespica committed:
adjust driver/AuthManager

Thespica committed:
add drivers and modify HugeClient

Thespica committed:
add drivers and modify HugeClient

Thespica committed:
replace particial group by role?

Thespica committed:
replace particial group by role?

Thespica committed:
modify for GraphsAPI ans GraphsManager

Thespica committed:
modify for GraphsAPI ans GraphsManager

Thespica committed:
fix func return type

Thespica committed:
fix func return type

Thespica committed:
modify traverser related tests

Thespica committed:
modify traverser related tests

Thespica committed:
adjust part of Treverser related tests

Thespica committed:
adjust part of Treverser related tests

Thespica committed:
fix decerialization

Thespica committed:
fix decerialization

Thespica committed:
fix test for particial traverser api

Thespica committed:
fix test for particial traverser api

Thespica committed:
test ci

Thespica committed:
test ci

Thespica committed:
patch for CustomizedPathsRequest

Thespica committed:
patch for CustomizedPathsRequest

Thespica committed:
test ci

Thespica committed:
test ci

Thespica committed:
modify test detels

Thespica committed:
modify test detels

Thespica committed:
temp for auth modify

Thespica committed:
temp for auth modify

imbajin committed:
Merge branch 'master' into client-support-gs

imbajin committed:
Merge branch 'master' into client-support-gs

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need' into hubble2.0

Thespica committed:
add structures of Role and UserManager

Thespica committed:
add structures of Role and UserManager

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need' into hubble2.0

Thespica committed:
GraphManager-getTypeCounts

Thespica committed:
GraphManager-getTypeCounts

Thespica committed:
GraphManager-updateVertexProperty

Thespica committed:
GraphManager-updateVertexProperty

Thespica committed:
GraphManager-updateEdgeProperty

Thespica committed:
GraphManager-updateEdgeProperty

Thespica committed:
User struct add fields

Thespica committed:
User struct add fields

Thespica committed:
Belong struct add fields

Thespica committed:
Belong struct add fields

Thespica committed:
Target struct add fields

Thespica committed:
Target struct add fields

Thespica committed:
Login struct add fields

Thespica committed:
Login struct add fields

Thespica committed:
Access struct add graph space field

Thespica committed:
Access struct add graph space field

Thespica committed:
SchemaManager add builder related methods

Thespica committed:
SchemaManager add builder related methods

Thespica committed:
SchemaManager add updatePropertyKey method

Thespica committed:
SchemaManager add updatePropertyKey method

Thespica committed:
SchemaManager add getGroovySchema method

Thespica committed:
SchemaManager add getGroovySchema method

Thespica committed:
patch for PropertyKey builder

Thespica committed:
patch for PropertyKey builder

Thespica committed:
Task add type() and PRIORITY field

Thespica committed:
Task add type() and PRIORITY field

Thespica committed:
CrosspointsRequest make some fields public

Thespica committed:
CrosspointsRequest make some fields public

Thespica committed:
fix rebase

Thespica committed:
update copyright

Thespica committed:
add dependency: hg-pd-client-1.5.0 and hg-pd-grpc-1.5.0

Thespica committed:
add constructors for RestClient

Thespica committed:
AuthElement add setId method

Thespica committed:
upgrade java-client-ci java version to 11

Thespica committed:
add id method for vertexLavel builder

Thespica committed:
add id method for EdgeLabel builder

Thespica committed:
add id method for IndexLabel builder

Thespica committed:
restore url for Target struct

Thespica committed:
ask messege to drop and clear graph in GraphsApi

Thespica committed:
fix format error

Thespica committed:
upgrade codeql-analysis ci's java version to 11

Thespica committed:
upgrade mockito version to support java 11

Thespica committed:
temporarily adjust hubble to pass ci

Thespica committed:
fix KneighborApiTest

Thespica committed:
ignore login related test due to server's unsupportness

Thespica committed:
ignore login test

Thespica committed:
regenerate known dependencies

Thespica committed:
AuthManager add ManagerAPI and related methods

Thespica committed:
add license header

Thespica committed:
remove unused override function for ManagerAPI

zyxxoo commented on hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/HugeGraphHubble.java:
```diff
@@ -27,28 +27,34 @@
     public static void main(String[] args) {
+        System.out.println("user.dir ==> " + System.getProperty("user.dir"));
         initEnv();
+        TimeZone.setDefault(TimeZone.getTimeZone(ZoneOffset.of("+8")));
```
the zone default +8 should set by config, because the repo not only use in china

zyxxoo replied:
some. infra cant auto transform zone, so maybe wo shouldn't set default zone

zyxxoo commented on hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/config/IngestionProxyServlet.java:
```diff
@@ -0,0 +1,72 @@
+                                     HttpServletResponse servletResponse,
+                                     HttpRequest proxyRequest) throws IOException {
+        String username =
+                (String) servletRequest.getSession().getAttribute("username");
```
magic

zyxxoo commented on hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/controller/auth/LoginController.java:
```diff
@@ -0,0 +1,79 @@
+    @PostMapping("/login")
+    public Object login(@RequestBody Login login) {
+        // Set Expire: 1 Month
+        login.expire(60 * 60 * 24 * 30);
```
the expire time should been configed

zyxxoo commented on hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/controller/op/AuditController.java:
```diff
@@ -0,0 +1,91 @@
+//
+//@RestController
+//@RequestMapping(Constant.API_VERSION + "audits")
+//public class AuditController extends BaseController {
```
delete the file

zyxxoo commented on hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/controller/op/MonitorController.java:
```diff
@@ -0,0 +1,57 @@
+    @GetMapping
+    public Object monitor() {
+        String monitorURL = null;
+        // Get monitor.url from system.env
```
1. first env
2. config file

Thespica committed:
fix format for AuthManager

FrostyHec committed:
adapt import to internal-client

FrostyHec committed:
hide resource and role manage modules.

FrostyHec committed:
fix(hubble): hide resource and role manage modules

FrostyHec committed:
removing backend support of RoleController,AccessController, BelongController,TargetController

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/client-support-gs' into hubble2.0-auth-simplified

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need' into hubble2.0-auth-simplified

FrostyHec committed:
removing backend support of RoleController,AccessController, BelongController,TargetController

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need-rebase-master' into hubble2.0-auth-simplified

# Conflicts:
#	.github/workflows/client-ci.yml
#	hugegraph-client/pom.xml
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/graphs/GraphsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/ip/WhiteIpListAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/kvstore/KvStoreAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/task/ComputerDisAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/AdamicAdarAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/EgonetAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/ResourceAllocationAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/SameNeighborsBatchAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/VariablesAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/vermeer/VermeerAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/client/RestClient.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/ComputerManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/GraphsManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/HStoreManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/PDManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/SchemaTemplateManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/ServiceManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/VermeerManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/WhiteIpListManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/factory/PDHugeClientFactory.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/auth/Role.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/auth/Target.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/auth/UserManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/CountResponse.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/Egonet.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/EgonetRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/PathOfVertices.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/PathWithMeasure.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/Prediction.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/RanksWithMeasure.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighbors.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighborsBatch.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighborsBatchRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/Steps.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/GraphsApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/AdamicAdarAPITest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KneighborApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/SameNeighborsBatchApiTest.java
#	hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/service/algorithm/OltpAlgoService.java

FrostyHec committed:
fix metrics client dependency

Thespica committed:
AuthManager add method createBatch

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need-rebase-master' into hubble2.0-auth-simplified

FrostyHec committed:
Adapt GraphsService to client

FrostyHec committed:
refactor(hubble): adapt GraphsService to client

Thespica committed:
translate Chinese comment into English

Thespica committed:
update client ci's server & allow running login test

Thespica committed:
remove useless code of JaccardSimilarityAPI

Thespica committed:
add comment to CountAPI

Thespica committed:
remove useless code of RingsAPI

Thespica committed:
add comment to PDHugeClientFactory

Thespica committed:
remove temp impl of encode

Thespica committed:
upgrade client's ci runner to ubuntu-latest

Thespica committed:
translate Chinese comment to English

Thespica committed:
pull and translate

Thespica committed:
remove useless comment

Thespica committed:
refine message str paramiters

Thespica committed:
wrapper for delete and clear graphs

FrostyHec committed:
Merge remote-tracking branch 'refs/remotes/thespica/patch-hubble-need-rebase-master' into hubble2.0-auth-simplified

akeyz commented:
你好，请问新版本上线了吗？这个问题得到修复了嘛？

akeyz commented:
> > 你好，请问新版本上线了吗？这个问题得到修复了嘛？
> 
> 还没有, 我们尽快推进恢复这个 PR, 已经 90% 进度了 (也有一个可直接运行的二进制包)

你好，这个可运行的二进制包在哪边可以下载到？方便给一下链接么？谢谢。

FrostyHec committed:
Merge remote-tracking branch 'origin/master' into hubble2.0-auth-simplified

# Conflicts:
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/auth/ManagerAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/graphs/GraphsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/kvstore/KvStoreAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/task/ComputerDisAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/AllShortestPathsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/CrosspointsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/NeighborRankAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/PathsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/PersonalRankAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/RaysAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/ShortestPathAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/ComputerManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/HStoreManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/HugeClient.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/PDManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/SchemaTemplateManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/ServiceManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/TraverserManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/VermeerManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/WhiteIpListManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/serializer/direct/HBaseSerializer.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/schema/PropertyKey.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/EgonetRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/KneighborRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/KoutRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighborsBatchRequest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/FusiformSimilarityApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KoutApiTest.java
#	hugegraph-hubble/hubble-be/src/main/java/org/apache/hugegraph/service/algorithm/OltpAlgoService.java

FrostyHec committed:
Update dependencies

FrostyHec committed:
Merge branch 'apache:master' into hubble2.0-auth-simplified

FrostyHec committed:
Fixing Dependency issues between Hubble and Client.

FrostyHec committed:
Merge remote-tracking branch 'upstream/master' into hubble2.0-auth-simplified

FrostyHec committed:
Fix token issue to match backend

imbajin committed:
Merge branch 'master' into pr/632

imbajin committed:
fix: remove duplicate client() method in BaseApiTest

Eliminated a redundant protected client() method definition in BaseApiTest to clean up the code and avoid confusion.

imbajin committed:
refactor(loader): improve code readability and formatting

Refactored several code blocks in HugeGraphLoader.java and JDBCFetcher.java to improve readability by adding line breaks, consistent indentation, and clearer formatting. No functional changes were made.

imbajin committed:
chore(hubble): improve Node.js build caching and update Sass dependency

Added Node modules caching to GitHub Actions workflow for faster CI builds. Updated the Maven build to use yarn with --frozen-lockfile and --prefer-offline for more reliable installs. Replaced 'node-sass' with 'sass' in frontend dependencies to ensure compatibility and maintainability.

imbajin committed:
fix(hubble): merge conflict and add yarn resolutions

Resolved a merge conflict in convertStringToJSON.js and removed unrelated Java code. Added a 'resolutions' field for 'pretty-format' in package.json to address dependency versioning. yarn.lock added to track dependencies.

imbajin committed:
chore(hubble): update Node.js version to 18.20.8 across project

Updated Node.js version requirements and setup from 16.x to 18.20.8 in CI workflow, documentation, and build configuration. Also removed the 'resolutions' field for 'pretty-format' from package.json.

imbajin committed:
chore(hubble): set CI env vars and increase Node memory in build

Added CI and NODE_OPTIONS environment variables to both the GitHub Actions workflow and the Maven build configuration. This sets CI to false and increases Node.js memory allocation to 4096MB to improve build stability and provide clearer environment diagnostics.

imbajin committed:
feat: init serena onboarding & project memory files

Introduces the .serena directory with onboarding documentation, project architecture, code style, workflows, and related memory files for HugeGraph Toolchain.

These files provide guidelines and quick start information for new developers, covering project overview, design patterns, code conventions, testing, and development workflows.

imbajin committed:
Merge branch 'master' into hubble2.0

FrostyHec committed:
Merge remote-tracking branch 'origin/master' into hubble2.0-auth-simplified

# Conflicts:
#	hugegraph-client/assembly/travis/install-hugegraph-from-source.sh

FrostyHec committed:
feat(client): introduce args for optional checking

FrostyHec committed:
Introduce args for optional checking when constructing HugeClientBuilder

FrostyHec committed:
fix dependency issues for correct construction

FrostyHec committed:
Merge remote-tracking branch 'origin/hubble2.0-auth-simplified' into hubble2.0-auth-simplified

FrostyHec committed:
Merge remote-tracking branch 'origin/hubble2.0' into hubble2.0

# Conflicts:
#	.github/workflows/client-ci.yml
#	hugegraph-client/assembly/travis/install-hugegraph-from-source.sh
#	hugegraph-client/pom.xml
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/graph/GraphMetricsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/graphs/GraphsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/ip/WhiteIpListAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/kvstore/KvStoreAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/task/ComputerDisAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/AdamicAdarAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/AllShortestPathsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/CountAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/CrosspointsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/EgonetAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/JaccardSimilarityAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/NeighborRankAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/PathsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/PersonalRankAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/RaysAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/ResourceAllocationAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/RingsAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/SameNeighborsBatchAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/ShortestPathAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/traverser/VariablesAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/api/vermeer/VermeerAPI.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/client/RestClient.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/ComputerManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/GraphManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/GraphsManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/HStoreManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/HugeClient.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/PDManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/SchemaTemplateManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/ServiceManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/TraverserManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/VermeerManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/WhiteIpListManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/driver/factory/PDHugeClientFactory.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/auth/Role.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/auth/UserManager.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/space/HStoreNodeInfo.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/CountResponse.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/Egonet.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/EgonetRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/KneighborRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/KoutRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/PathOfVertices.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/PathWithMeasure.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/Prediction.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/RanksWithMeasure.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighbors.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighborsBatch.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/SameNeighborsBatchRequest.java
#	hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/Steps.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/GraphsApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/AdamicAdarAPITest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/FusiformSimilarityApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KneighborApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KoutApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/SameNeighborsBatchApiTest.java
#	hugegraph-client/src/test/java/org/apache/hugegraph/functional/HugeClientHttpsTest.java

FrostyHec committed:
Merge remote-tracking branch 'upstream/master' into hubble2.0

# Conflicts:
#	hugegraph-loader/src/main/java/org/apache/hugegraph/loader/HugeGraphLoader.java

FrostyHec committed:
update yarn.lock

FrostyHec committed:
Merge branch 'hubble2.0-auth-simplified' into hubble2.0

# Conflicts:
#	hugegraph-hubble/hubble-dist/pom.xml
#	hugegraph-hubble/hubble-fe/package-lock.json
#	hugegraph-hubble/hubble-fe/package.json
#	hugegraph-hubble/hubble-fe/yarn.lock

FrostyHec committed:
update yarn.lock

FrostyHec committed:
refactor(client): add graphRequired option in HugeClientBuilder

FrostyHec committed:
refactor(client): add graphRequired option in HugeClientBuilder (#718)

* refactor(client): replace skipRequiredChecks bool param with chainable graphRequired() method

- Move url/graph validation from constructor to build() for true null support
- Replace boolean constructor param with fluent .graphRequired(false) chain method
- Rename field to graphRequired (default true) for clearer semantics
- Clean up HugeClient javadoc: remove redundant class/constructor comments

---------

Co-authored-by: imbajin <jin@apache.org>

Yeaury committed:
fix(hubble-be): Fix bugs related to updating user passwords.

Yeaury committed:
fix(hubble-be): Fix NPE when deleting a graph.

Yeaury committed:
fix(client): fix the issue where the graph space is null.

Commit:
fix(hubble): audit binary dist packaging

- source Hubble binary legal files from release docs
- remove runtime residue before creating the Hubble tarball
- add a dist inventory check for required paths and blocked artifacts
- support foreground startup without stale daemon output redirection

Commit:
fix(hubble): harden binary dist audit

- validate packaged legal files are present and non-empty
- remove frontend source maps from Hubble binary packaging
- record optional checksum and signature verification status
- scan native-bearing jars from one extracted distribution tree

imbajin committed:
fix(hubble): harden runtime smoke gates

- fill standalone loader connection endpoint

- validate runtime loader and shortestPath flows

- keep Hubble alive across wrapper UI checks

- add Playwright dependency for browser smoke

imbajin committed:
fix(hubble-be): enforce authz routes

- wire login interceptor and proxy auth boundaries
- normalize graphspace auth controller responses
- add role, access, target, and belong service guards
- cover auth tokens and route registration with unit tests

imbajin committed:
fix(hubble): stabilize graph analysis ui

- normalize Gremlin and Vermeer responses
- localize graph analysis and async task labels
- preserve graph result shape for visualization
- add result formatting and graph utility tests

imbajin committed:
fix(ci): enable auth for hubble smoke server

- enable StandardAuthenticator for hubble server1 smoke fixture
- initialize server1 admin password for noninteractive CI
- align server1 login credentials with Hubble smoke defaults
- keep auth fixture scoped to localhost travis server config

imbajin committed:
fix(hubble): encode schema export filename

- sanitize schema export download filename
- add UTF-8 filename star disposition
- use octet-stream for schema download
- cover CRLF and non-ASCII graph names

imbajin committed:
fix(hubble): harden graph JSON import

- validate upload size and JSON structure before materializing import data
- precheck vertices, edges, endpoints, duplicate ids, and label matches before writes
- ignore imported edge ids for create semantics and support staged vertex links
- rollback created edges and vertices on write failure
- add focused GraphService import tests to UnitTestSuite

imbajin committed:
fix(hubble): degrade non-PD navigation

- add product mode helpers for non-PD routing
- keep source and task visible outside PD mode
- redirect PD-only routes away from non-PD pages
- hide graph create/default/clone entries in non-PD
- record Chrome and local validation evidence

imbajin committed:
fix(hubble-fe): localize task edit

- move task edit steps, forms, mappings, and schedule text into i18n resources
- pass localized validation messages into shared task edit rules
- remove obsolete task edit duplicate component files
- verify task edit i18n with focused tests and frontend build

imbajin committed:
fix(hubble-fe): localize canvas controls

- move shared graph canvas toolbar, menu, filter drawer, tooltip, legend and task navigation labels into analysis.canvas i18n keys
- add focused canvas i18n coverage for new English resource keys
- verify gremlin and algorithm canvas surfaces with Chrome evidence

imbajin committed:
fix(hubble): widen loader file path

- widen file_mapping.path to VARCHAR(2048)
- add startup migration for legacy H2/MySQL/MariaDB schemas
- cover fresh schema and legacy H2 deep path inserts
- register the regression test in UnitTestSuite

imbajin committed:
fix(hubble-fe): declare qs dependency

- declare qs for existing API imports

- keep yarn and npm lock selectors aligned

- preserve dependency-only commit boundary

imbajin committed:
fix(hubble): pass loader connection options

- map Hubble loader Basic credentials to password with token fallback
- preserve PD graphspace and routing fields for embedded loader
- keep direct server loader smoke independent from PD peers
- add focused loader option tests and live smoke boundary marker

imbajin committed:
fix(hubble): localize async task constants

- Move async task type, status, and action labels to shared i18n key maps.

- Reuse a common translation helper in async task detail.

- Add full en-US and zh-CN resource coverage tests for dynamic keys.

- Verify with Chrome evidence, focused Jest, i18n check, and FE build.

imbajin committed:
fix(hubble): localize edge tooltip labels

- Move edge label type display text into i18n resources.
- Translate shared graph tooltip edge tags at render time.
- Cover edge label type constants with focused i18n tests.
- Verify with Chrome, focused Jest, i18n check, eslint, and FE build.

imbajin committed:
fix: remove empty catch blocks

- replace HubbleConfig resource lookup empty catch with explicit null check
- restore interrupt state and fail TaskApiTest sleeps on interruption
- add EmptyCatchTest and include it in Hubble UnitTestSuite
- verify with JDK 11 EmptyCatchTest, UnitTestSuite, client test-compile

imbajin committed:
fix(hubble): guard protected routes

- add a protected route wrapper for layout business routes
- preserve redirect targets for login return flow
- catch Topbar auth status request failures
- add route guard and Topbar request regression tests

imbajin committed:
fix(hubble): use gremlin client for shortest path

- Route shortestPath through the gremlin-capable client

- Add a controller unit test for the client selection

- Preserve token-client behavior for unrelated OLTP APIs

- Validate with focused tests, unit suite, package and PD smoke

imbajin committed:
docs(hubble): add api cleanup plan

- split backend and frontend contract slices\n- define red-green verification commands\n- preserve client compatibility boundary\n- add integrated acceptance audit

imbajin committed:
refactor(hubble): use canonical graph api calls

- send graph updates with PUT JSON
- read the canonical default graph resource
- remove default GraphSpace facade calls
- hide unsupported default GraphSpace mutation

imbajin committed:
feat(hubble): require strong graph clear confirmation

- show graph and deletion scope
- require exact graph-name confirmation
- retain errors without closing the dialog
- document temporary data-only clear semantics

imbajin committed:
fix(ci): install playwright chromium

- cache Playwright browsers by frontend lockfile
- install Chromium with required system dependencies
- keep browser provisioning separate from frontend checks

imbajin committed:
docs(hubble): refresh active TODO state

- remove recently completed TODO entries
- keep ten unfinished tasks grouped by priority
- record the current CI failure boundary
- normalize tracked evidence for EditorConfig

imbajin committed:
docs(hubble): defer frontend CI cleanup

- keep backend CI on the primary release path\n- move frontend CI legacy cleanup to final priority\n- preserve real Chrome acceptance as the product gate

imbajin committed:
chore(hubble): complete license headers

- add canonical ASF headers to Java and JavaScript tests
- cover Hubble browser acceptance helper files
- record the failed PR license-header check
- preserve all existing runtime and test behavior

imbajin committed:
docs(hubble): record final validation

- record implementation head workflow results
- capture independent review and re-review evidence
- mark the release-readiness checklist complete

imbajin committed:
chore(hubble): align core versions and style

1. replace hardcoded hg-pd-client and hugegraph-core versions with a unified `hugegraph.core.version` (1.7.0) property.
2. update root `lombok.version` from 1.18.8 to 1.18.32 for newer toolchain compatibility.
3. migrate `-Xmaxerrs` compiler configuration to `compilerArgs` in `hubble-be/pom.xml`.
4. clean up wildcard imports, spacing, and line wrapping across hubble-be controllers, services, entities, and utilities for consistent code style.

imbajin committed:
chore(hubble): cover schema view contracts

- verify schema view success mapping for property types\n- verify missing property keys fail with the existing contract\n- confirm test sensitivity with a temporary focused mutation\n- pass the full 121-test backend unit suite

imbajin committed:
fix(ci): cache exact server package

- key the HugeGraph server cache by resolved commit SHA\n- reuse only the tarball built for that immutable SHA\n- preserve checkout verification and source build on misses\n- validate shell syntax, YAML parsing, and cache-hit behavior

imbajin committed:
docs(hubble): refine quality decisions

- record CI=true package and distribution audit evidence\n- capture the effective checkstyle gate mutation proof\n- split the React pin from external source-map warnings\n- preserve high-risk dependency boundaries

imbajin committed:
fix(hubble): restore strict frontend builds

- pin React and ReactDOM to the stable 18.2 runtime\n- remove the unused conflicting npm lockfile\n- run Maven frontend packaging with CI enabled\n- verify lint, 148 Jest tests, browser UI, and dist audit

imbajin committed:
docs(hubble): record final local gates

- capture fresh FE, BE, linked, and package results
- clarify CI quota as delayed external evidence
- refresh current HEAD and worktree boundaries
- correct the Maven toolchain lesson

imbajin committed:
docs(hubble): close review evidence

- record independent review and re-review outcomes
- capture the third-party inventory CI fix
- add candidate-only reflection lessons
- leave only final real CI evidence pending

imbajin committed:
style(hubble): purge legacy frontend hubble-fe-old and update dependency configurations

imbajin committed:
fix(hubble): extend active session timeout

- set the servlet idle timeout to 48 hours
- rely on standard access-based inactivity refresh
- add a contract test for the packaged Hubble config

Thespica committed:
feat(hubble): support hubble 2.0 alpha client & backend base

Co-authored-by: yeaury <yeaury@qq.com>

imbajin committed:
chore: consolidate codex task tracking

- keep one concise global follow-up list
- stop tracking goal-specific process and evidence files
- ignore local task artifacts by default
- preserve completed audit evidence locally

imbajin committed:
docs: consolidate remaining hubble work

- capture the unfinished UI and UX refactor
- keep completed legacy and hardening tasks excluded
- preserve detailed execution steps in local task records

imbajin committed:
chore(hubble): correct README references and update gitignore

imbajin committed:
fix(hubble): harden request security

- disable Kubernetes token endpoints\n- validate pagination and close request clients\n- secure CORS and sanitize exception logging\n- clean up graph listeners and loading state\n- add backend security regression coverage

imbajin committed:
fix(hubble): throttle failed logins

- add configurable exponential backoff after repeated failures
- return HTTP 429 with Retry-After and backend warnings
- serialize concurrent attempts per principal and client address
- show a deduplicated frontend warning and cover regressions

imbajin committed:
fix(hubble): scope loader resources

- bind loader entities to graphspace, graph, and job scopes
- protect upload, batch, mutation, and deletion paths
- add scoped access regression coverage

imbajin committed:
chore(hubble): trim legacy build debt

- remove unused secondary request wrapper
- use default Node and Yarn download sources
- align Kotlin runtime dependency versions
- reduce expected parser log noise and mark date migration

imbajin committed:
fix: add standard ASF license headers and fix checkstyle issues

Files were using a short-form license header that fails the license-eye
check. Replace with the full standard Apache 2.0 header consistent
with the rest of the project.

Also fix all remaining Checkstyle violations in toolchain modules
including line length, indentation, operator wrapping, and support for
SuppressWarnings filter.

Files fixed (License):
- hubble-be/.../exception/LoginThrottledException.java
- hubble-be/.../service/auth/LoginAttemptGuard.java
- hubble-be/.../unit/LoaderScopeControllerTest.java
- hubble-be/.../unit/LoginAttemptGuardTest.java
- hubble-be/.../unit/PriorityFixTest.java
- hubble-fe/src/api/throttleWarning.js
- hubble-fe/src/components/RouteErrorBoundary/index.js

Files fixed (Checkstyle):
- tools/checkstyle.xml
- hugegraph-client/.../driver/HugeClientBuilder.java
- hugegraph-loader/.../loader/HugeGraphLoader.java
- hugegraph-loader/.../loader/task/TaskManager.java
- hugegraph-loader/.../loader/executor/LoadOptions.java

imbajin committed:
feat(hubble): add safe Text2GQL preview

- add a peer natural-language query tab
- block all placeholder execution and network paths
- recover rejected query and async task requests
- cover zero-call and failure contracts with tests

imbajin committed:
fix(hubble): restore algorithm interactions

- forward only rc-collapse control props across wrappers
- normalize rejected algorithm requests for form recovery
- preserve success, business-error, and auth contracts
- cover all algorithm panels and shared API boundaries

imbajin committed:
fix(hubble): harden result view switching

- project BigInt and JSONbig values safely for JSON display
- render table cells without lossy quoting or precision changes
- prevent hidden graph fitting and duplicate layout registration

imbajin committed:
fix(hubble): preserve exported graph ids

- serialize large graph ids as exact decimal strings
- remove the JSON blob BOM and verify parse round trips
- expose export actions through keyboard-operable menu items

imbajin committed:
fix(hubble): open export menu on click

- make the export button open its dropdown on click
- preserve Ant Design keyboard-enabled menu item actions
- verify the explicit trigger contract with targeted tests

imbajin committed:
fix(hubble): clarify query recovery state

- persist a controlled reason for synchronous Cypher failures
- keep async submission and execution failure semantics distinct
- close favorite popovers when switching query languages
- cover sync, async, and tab-transition regression paths

imbajin committed:
fix(hubble): recover import task options

- keep option loading failures visible with scoped retries
- clear stale graph and source-field form values
- ignore late responses after context changes
- localize recovery states and cover dual-mode contracts

imbajin committed:
fix(hubble): align graph mutation contracts

- scope clear copy and symbols to graph data
- gate incompatible clone actions with accessible reasons
- preserve destructive confirmation and refresh behavior
- lock API and action contracts with regression tests

imbajin committed:
fix(hubble): recover system workspace states

- preserve loading, failure and retry contracts across system pages
- prevent stale profile, task and metadata responses from leaking
- replace click-only controls with keyboard-accessible actions
- localize login, recovery and unknown-route semantics
- add focused accessibility and race regression coverage

imbajin committed:
fix(hubble): recover schema capabilities

- expose graph schema export with identity-safe recovery
- distinguish Meta loading, empty and failure states
- recover vertex and edge detail requests without stale writes
- clarify optional template application during graph creation
- make graph creation and card actions keyboard accessible

imbajin committed:
fix(hubble): restore schema export compatibility

- rebuild Groovy from structured Schema responses
- preserve nested userdata and quoted separators
- keep legacy Server output unchanged
- verify real PD export and replay
- add compatibility regression coverage

imbajin committed:
fix(hubble): resolve accessibility and mock errors in UI tests

imbajin committed:
fix(hubble): resolve eslint, missing license headers, i18n keys, and path traversal security warnings

imbajin reviewed (approved):
Merge it first, then enhance UI/UX/CI later, congrats:)
